go/vt/vtgate/schema/update_controller.go

```diff
@@ -119,7 +119,7 @@ func (u *updateController) getItemFromQueueLocked() *discovery.TabletHealth {
 
 func (u *updateController) add(th *discovery.TabletHealth) {
 	// For non-primary tablet health, there is no schema tracking.
-	if th.Tablet.Type != topodatapb.TabletType_PRIMARY {
+	if th.Target.TabletType != topodatapb.TabletType_PRIMARY {
```

Comment:
This is the "second" fix where we look at the target instead of the current tablet type, the same approach is used in VTTablet